numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -2003,20 +2003,40 @@ array_scalar(PyObject *NPY_UNUSED(ignored), PyObject *args, PyObject *kwds)
     int alloc = 0;
     void *dptr;
     PyObject *ret;
-
+    PyObject *base = NULL;
 
     if (!PyArg_ParseTupleAndKeywords(args, kwds, "O!|O:scalar", kwlist,
                 &PyArrayDescr_Type, &typecode, &obj)) {
         return NULL;
     }
     if (PyDataType_FLAGCHK(typecode, NPY_LIST_PICKLE)) {
-        if (!PySequence_Check(obj)) {
-            PyErr_SetString(PyExc_TypeError,
-                            "found non-sequence while unpickling scalar with "
-                            "NPY_LIST_PICKLE set");
+        if (typecode->type_num == NPY_OBJECT) {
+            if (DEPRECATE(
+                    "Attempting to unpickling a numpy object scalar; "
+                    "object scalars should never be created. If this was a "
```

Comment:
```suggestion
            /* Deprecated 2020-11-24, NumPy 1.20 */
            if (DEPRECATE(
                    "Unpickling a scalar with object dtype is deprecated. "
                    "Object scalars should never be created. If this was a "
```

Reply:
Thanks, updated. I could push out the commit to make backporting easier if we do not want to include it in a possible backport.